scripts/exregional_get_extrn_mdl_files.sh

```diff
@@ -175,9 +174,95 @@ ${cmd}
+# 2 days before).  In the DTC Ensemble Task, we use these GEFS files
+# (instead of the GFS ones) with perturbations added (in another script)
+# to calculate the initial conditions.
+#
+#-----------------------------------------------------------------------
+#
+# Try to obtain the GEFS member number from the name of the experiment.
+# This is just the two digits within the experiment name.
+#
+gefs_mem=""
+gefs_mem=$( printf "%s" "${EXPT_SUBDIR}" | $SED -n -r -e "s/^GEFS_mem([0-9]{2}).*$/\1/p" )
+
+if [ ! -z "${gefs_mem}" ] && [[ ! "${gefs_mem}" =~ ^[0-9]{2}$ ]]; then
+  print_err_msg_exit "\
+The variable \"gefs\"_mem must either be a 2-digit string representing the 
+GEFS member number or an empty string; otherwise, something is wrong:
+  gefs_mem = \"${gefs_mem}\""
+fi
+#
+# If gefs_mem is not empty, it means the current workflow is for a GEFS
+# individual member (as opposed to being the RRFS analog).  In that case,
+# perform the external model file replacement (replace GFS files with 
+# GEFS files).
+#
+if [ "${ICS_OR_LBCS}" = "ICS" ] && [ ! -z "${gefs_mem}" ]; then
+
+  print_info_msg "
+Replacing GFS external model IC files with GEFS external model IC files
+for GEFS member:
+  gefs_mem = \"${gefs_mem}\""
+
+  gfs_dir="${EXPTDIR}/${CDATE}/FV3GFS/for_ICS"
+  gfs_fn="gfs.t00z.pgrb2.0p25.f000"
+  gfs_fp="${gfs_dir}/${gfs_fn}"
+  gfs_fp_orig="${gfs_fp}.orig"
+#
+# At least one of the GFS file (gfs_fp) and the renamed ("orig") version
+# of it (gfs_fp_orig) must exist.  If not, print out an error message
+# and exit.
+#
+  if [ ! -f "${gfs_fp}" ] && [ ! -f "${gfs_fp_orig}" ]; then
+    print_err_msg_exit "\
+The GFS external model file (gfs_fp) or its renamed version (gfs_fp_orig)
+must exist, but neither do:
+  gfs_fp = \"${gfs_fp}\"
+  gfs_fp_orig = \"${gfs_fp_orig}\""
+  fi
+#
+# If a renamed ("orig") GFS file already exists (e.g. due to a previous
+# call to this script), move it so it doesn't get overwritten in the
+# next step.
+#
+  check_for_preexist_dir_file "${gfs_fp_orig}" "rename"
+#
+# If the GFS file exists, rename it.  Note that it may not exist due to
+# a previous call to this script.
+#
+  if [ -f "${gfs_fp}" ]; then
+    mv_vrfy "${gfs_fp}" "${gfs_fp_orig}"
+  fi
+
+  gefs_cyc=$(date -d "${CDATE:0:8} -2 day" +%Y%m%d)
+  gefs_fn="${gefs_cyc}.gep${gefs_mem}.t18z.pgrb2.0p50.f030"
+  gefs_fp="${GEFS_STAGING_DIR}/${gefs_fn}"
+
+  print_info_msg "
+Copying GEFS file (gefs_fp) into experiment's external model IC file
+location (gfs_dir):
+  gefs_fp = \"${gefs_fp}\"
+  gfs_dir = \"${gfs_dir}\""
+  cp_vrfy "${gefs_fp}" "${gfs_dir}"
+
+  print_info_msg "
+Creating symlink (named identical to the GFS external model file (gfs_fn))
+to the GEFS file (gefs_fn):
+  gefs_fn = \"${gefs_fn}\"
+  gfs_fn = \"${gfs_fn}\""
+  ln_vrfy -fs --relative "${gefs_fn}" "${gfs_fn}"
+
+fi
+#
+#-----------------------------------------------------------------------
+#
 # Restore the shell options saved at the beginning of this script/function.
```

Comment:
Remember that these need to generate both an RRFS workflow and a GEFS workflow.  This new code section will only get activated when running a GEFS workflow (for a GEFS member).  For an RRFS workflow, the variable `gefs_mem` set on line ~191 will be an empty string because the experiment name will not have the form `GEFS_mem##` (the experiment name will be something like `RRFS_ICperts_[no]stoch`).  Thus, the condition in the if-statement on line ~205 will not true, and the code in it will not be executed (so the `HRRR` files obtained above will not be replaced).  But for a GEFS workflow, `gefs_mem` will be `01` or `02` or ..., so the if-statement will be true, and the GFS external model file will be replaced with the staged GEFS file.

Reply:
Thanks for the explanation!